standard/variables.md

```diff
@@ -901,3 +901,181 @@ variable_reference
+
+For any variable, the *safe_scope* of that variable is the scope where its value may be copied. A variable whose type is not a `ref struct` type is *safe-to-return* from the entire enclosing method. Its *safe_scope* is *caller_method*. Otherwise the following rules apply.
+
+> *Example*: The following code shows examples of the different *safe_scope* and *ref_safe_scope* values. The declarations show values of *safe_scope* for a referent to be the initializing expression for a `ref` variable. The examples show the values of *ref_safe_scope* for a reference return:
+>
+> ```csharp
+> public class C
+> {
+>     private int[] arr = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
+> 
+>     public ref int M1(ref int rv)
+>     {
+>         return ref rv; // rv is safe to ref return
+>     }
+>
+>     public ref int M2(int v)
+>     {
+>         return ref v; // error: v isn't safe to ref return
+>     }
+>
+>     public ref int M3(ref int rv)
+>     {
+>         int v = 5;
+> 
+>         return ref arr[v]; // arr[v] is safe to ref return
+>     }
+> 
+>     public void M4(int p) 
+>     {
+>         int v = 6;
+>         ref int rv = ref p; // safe scope of rv is block, ref safe scope of p is method
+>         ref int rv2 = ref v; // safe scope of rv2 is block, ref safe scope of v is block
+>         ref int rv3 = ref arr[v]; // safe scope of rv3 is block, ref safe scope of arr[v] is caller method
+>     }
+> }
+> ```
+>
+> *end example.*
+
+### §ref-span-safety-locals Local variable safe scope
+
+For a local variable `v`:
+
+- If `v` is a `ref` variable, its *ref-safe-scope* is taken from the *ref-safe-scope* of its initializing expression.
+- Otherwise its *ref-safe-scope* is the scope in which it was declared.
+
+The *safe-scope* of an expression that designates the use of a local variable is determined as follows:
+
+- A value whose type is not a `ref struct` type is *safe-to-return* from the entire enclosing method. Its *safe-scope* is the *caller-scope*.
+- If the variable is an iteration variable of a `foreach` loop, then the variable's *safe-scope* is the same as the *safe-scope* of the `foreach` loop's expression.
+- A local of `ref struct` type which is uninitialized at the point of declaration is *safe-to-return* from the entire enclosing method. Its *safe-scope* is the *calling method*.
+- Otherwise the variable's type is a `ref struct` type, and the variable's declaration has an initializer. The variable's *safe-scope* is the same as the *ref-safe-scope* of its initializer.
+
+### §ref-span-safety-parameters Parameter escape scope
+
+For a formal parameter `p`:
+
+- If `p` is a `ref`, or `in` parameter, its *ref-safe-scope* is the *caller-scope*. It is *safe-to-return* by ref. If `p` is an `in` parameter, it can't be returned as a writable `ref` but can be returned as `ref readonly`.
+- If `p` is an `out` parameter, its *ref-safe-to-escape-scope* is the *current-method*. It isn't *safe-to-return* by ref.
```

Comment:
I think this is the first time ref-safe-to-escape has been mentioned, and I see various other references to safe-to-escape-scope which hasn't been defined.

If we could make sure that every term has a definition in bold and italic, it'll be easier to make sure we never use anything that's not defined.